include/beman/inplace_vector/inplace_vector.hpp

```diff
@@ -0,0 +1,1034 @@
+with the conditions of Sections 4(a), 4(b) and 4(d) of the License.
+
+In addition, if you combine or link compiled forms of this Software with
+software that is licensed under the GPLv2 ("Combined Software") and if a
+court of competent jurisdiction determines that the patent provision (Section
+3), the indemnity provision (Section 9) or other Section of the License
+conflicts with the conditions of the GPLv2, you may retroactively and
+prospectively choose to deem waived or otherwise exclude such Section(s) of
+the License, but only in their entirety and only with respect to the Combined
+Software.
+ */
+#include <algorithm> // for rotate, equals, move_backwards, ...
+#include <array>
+#include <concepts>   // for lots...
+#include <cstddef>    // for size_t
+#include <cstdint>    // for fixed-width integer types
+#include <functional> // for less and equal_to
+#include <iterator>   // for reverse_iterator and iterator traits
+#include <limits>     // for numeric_limits
+#include <new>        // for operator new
+#include <ranges>
+#include <stdexcept>   // for length_error
+#include <stdio.h>     // for assertion diagnostics
+#include <type_traits> // for aligned_storage and all meta-functions
+
+// Optimizer allowed to assume that EXPR evaluates to true
+#define __IV_ASSUME(__EXPR)                                                    \
+  static_cast<void>((__EXPR) ? void(0) : __builtin_unreachable)
+
+// Assert pretty printer
+#define __IV_ASSERT(...)                                                       \
+  static_cast<void>((__VA_ARGS__)                                              \
+                        ? void(0)                                              \
+                        : ::beman::__iv_detail::__assert_failure(              \
+                              static_cast<const char *>(__FILE__), __LINE__,   \
+                              "assertion failed: " #__VA_ARGS__))
+
+// Assert in debug, assume in release.
+// #ifdef NDEBUG
+// #define __IV_EXPECT(__EXPR) __IV_ASSUME(__EXPR)
+// #else
+// #define __IV_EXPECT(__EXPR) __IV_ASSERT(__EXPR)
+// #endif
+
+// TODO River: Disabled temporarily
+#define __IV_EXPECT(__EXPR)
+
+/*
+// BUGBUG workaround for libstdc++ not providing from_range_t / from_range yet
+namespace std {
+#if defined(__GLIBCXX__) || defined(__GLIBCPP__)
+struct from_range_t {};
+inline constexpr from_range_t from_range;
+#endif
+}; // namespace std
+*/
+
+// TODO(River): using namespace std below, we need to specify
+// beman::from_range_t
+namespace beman {
+struct from_range_t {};
+inline constexpr from_range_t from_range;
+}; // namespace beman
+
+// Private utilities
+namespace beman::__iv_detail {
+
+template <class = void>
+[[noreturn]]
+static constexpr void __assert_failure(char const *__file, int __line,
+                                       char const *__msg) {
+  if consteval {
+    throw __msg; // TODO: std lib implementer, do better here
```

Comment:
well and even more is that this will be legal in c++26 after Poland https://www.open-std.org/jtc1/sc22/wg21/docs/papers/2024/p3068r6.html was voted in

Reply:
👌 thanks for the input, I am not a huge fan of this static asset infrastructure as well. 

I will likely remove them or refactor them with static assert